plugins/BotTracking/RecordBuilders/AIAssistantReports.php

```diff
@@ -0,0 +1,219 @@
+        $bindBase = $logAggregator->getGeneralQueryBindParams();
+
+        $sql = sprintf(
+            "SELECT `referer_name`, COUNT(*) AS `visits`
+             FROM %s AS `log_visit`
+             WHERE `referer_type` = %d
+               AND `referer_name` <> ''
+               AND %s
+             GROUP BY `referer_name`",
+            Common::prefixTable('log_visit'),
+            Common::REFERRER_TYPE_AI_ASSISTANT,
+            $where
+        );
+
+        $stmt   = Db::query($sql, $bindBase);
+        $result = [];
+
+        while ($row = $stmt->fetch()) {
+            /**
+             * @var array{visits: string|int, referer_name: string} $row
+             */
+            if (in_array($row['referer_name'], self::ASSISTANT_MAPPING)) {
+                $key          = (string)array_search($row['referer_name'], self::ASSISTANT_MAPPING);
+                $result[$key] = (int)$row['visits'];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param array<string, DataTable> $tables
+     * @param array<string, int> $visits
+     * @return void
+     */
+    private function populateTableForActionType(array $tables, int $actionType, LogAggregator $logAggregator, array $visits): void
+    {
+        $where    = $logAggregator->getWhereStatement('bot', 'server_time');
+        $bindBase = $logAggregator->getGeneralQueryBindParams();
+
+        $sql = sprintf(
+            "SELECT * FROM (SELECT bot.bot_name, log_action.name AS url, COUNT(*) AS requests
+             FROM %s AS bot
+             INNER JOIN %s AS log_action ON log_action.idaction = bot.idaction_url
+             WHERE log_action.name IS NOT NULL
+               AND log_action.name <> ''
+               AND log_action.type = %d
+               AND %s
+             GROUP BY bot.bot_name, url WITH ROLLUP) AS rollupQuery
+             ORDER BY bot_name, requests DESC, url",
+            BotRequestsDao::getPrefixedTableName(),
+            Common::prefixTable('log_action'),
+            $actionType,
+            $where
+        );
+
+        $resultSet  = Db::query($sql, $bindBase);
+        $actionRows = [];
+
+        while ($row = $resultSet->fetch()) {
+            /**
+             * @var array{requests: int, bot_name: ?string, url: ?string} $row
+             */
+            $label = $row['bot_name'];
+            $url   = $row['url'];
+
+            if (is_null($label)) {
+                continue;
+            }
+
+            if (!is_null($url)) {
+                $actionRows[] = $row;
+                continue;
+            }
+
+            $metrics = [
+                Metrics::COLUMN_REQUESTS          => $row['requests'],
+                Metrics::COLUMN_DOCUMENT_REQUESTS => $actionType === Action::TYPE_DOWNLOAD ? $row['requests'] : 0,
+                Metrics::COLUMN_PAGE_REQUESTS     => $actionType === Action::TYPE_PAGE_URL ? $row['requests'] : 0,
+                Metrics::COLUMN_ACQUIRED_VISITS   => $visits[$label] ?? 0,
```

Comment:
Isn't this double counting if an Assistant has at least one `TYPE_DOWNLOAD` and one `TYPE_PAGE_URL` result?